smartmeter.js

```diff
@@ -348,15 +348,23 @@ function main() {
     }
     if (adapter.config.obisFallbackMedium !== null && adapter.config.obisFallbackMedium !== undefined) {
         adapter.config.obisFallbackMedium = parseInt(adapter.config.obisFallbackMedium, 10);
-        if (adapter.config.obisFallbackMedium < 0 || adapter.config.obisFallbackMedium > 18 ) {
+        if (adapter.config.obisFallbackMedium < 0 || adapter.config.obisFallbackMedium > 18) {
             adapter.log.error('OBIS Fallback medium code ' + adapter.config.obisFallbackMedium + ' invalid, check your configuration!');
             return;
         }
         smOptions.obisFallbackMedium = adapter.config.obisFallbackMedium;
     }
-    adapter.log.debug('SmartmeterObis options: ' + JSON.stringify(smOptions));
-
-    smTransport = SmartmeterObis.init(smOptions, storeObisData);
+    adapter.log.debug('Smartmeter options: ' + JSON.stringify(smOptions));
+
+    if (smOptions.protocol === 'TicProtocol') {
+        const SmartmeterTic = require('./lib/ticcollector');
+        // There's no 'storeTicData' callback because why have that code in here when it can live in the lib?
+        smTransport = SmartmeterTic.init(adapter, setConnected, smOptions);
+    } else {
+        // OBIS by default
+        const SmartmeterObis = require('smartmeter-obis');
```

Comment:
@Apollon77 TBH I think it would be nice to move `storeObisData` & a simple init into their own `lib/obiscollector.js` file so that a) this `smartmeter.js` becomes smaller (it's just left with common config & stuff) and b) both collectors have their underlying mechanisms & related object creation in their own files. What do you think? I could easily do that but have no way to test it once done as only have a TIC device.